### PR TITLE
Docs: Notify that an entry point is required for creating a data plugin

### LINKED ISSUE #5456: Docs: examples on creating data plugin fail because they do not include registering with entry point
https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/data_types.html?highlight=#creating-a-data-plugin

The examples should explain that an entry point is required.

### PR DESCRIPTION
**Pull request by AhmedBasem20**
Fixes #5456

Walking through the examples [here](https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/data_types.html?highlight=#creating-a-data-plugin) will fail because the data entry point registering step is required but not mentioned.

I added a simple note for registering with an entry point.

### CODE REVIEW

File: docs/source/topics/data_types.rst [hunk 1]
```diff
@@ -1122,6 +1122,10 @@ Creating a new data type is as simple as creating a new sub class of the base :c
     class NewData(Data):
         """A new data type that wraps a single value."""
 
+.. note::
+
+    You must register a new data entry point. See `aiida.data <https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/plugins.html#aiida-data>`
```
Comment (sphuber) [inline docs/source/topics/data_types.rst:1127]:
```suggestion
    To be able to use the new ``Data`` plugin, it must be registered using an entry point.
    See :ref:`What is an entry point?<topics:plugins:entrypoints>` for details.
```

**Review by sphuber — CHANGES_REQUESTED** [review]
Thanks @AhmedBasem20 . Small suggestion for the phrasing and a fix for the link syntax, which was incorrect

**Comment by AhmedBasem20** [comment]
Thanks @sphuber for reviewing. The suggested changes are updated.

File: docs/source/topics/data_types.rst [hunk 2]
```diff
@@ -1122,6 +1122,11 @@ Creating a new data type is as simple as creating a new sub class of the base :c
     class NewData(Data):
         """A new data type that wraps a single value."""
 
+.. note::
+
+    To be able to use the new ``Data`` plugin, it must be registered using an entry point.
+	    See :ref:`What is an entry point?<topics:plugins:entrypoints>` for details.
```
Comment (sphuber) [inline docs/source/topics/data_types.rst:1128]:
```suggestion
    To be able to use the new ``Data`` plugin, it must be registered using an entry point.
    See :ref:`What is an entry point?<topics:plugins:entrypoints>` for details.
```

**Review by sphuber — APPROVED** [review]

**Pull request merged**